src/scripts/background.js

```diff
@@ -1410,20 +1410,29 @@ window.TogglButton = {
     TogglButton.$userState = state;
     const now = new Date();
     const inactiveSeconds = Math.floor((now - TogglButton.$lastWork.date) / 1000);
+    const updateLastWork = (date) => {
+      TogglButton.$lastWork = {
+        id: TogglButton.$curEntry.id,
+        date: date || now
+      };
+    };
 
-    if (TogglButton.$user && state === 'active' && TogglButton.$curEntry) {
+    if (TogglButton.$user && ['active', 'idle'].includes(state) && TogglButton.$curEntry) {
       // trigger discard time notification once the user has been idle for
       // at least 5min
+      const FIVE_MINUTES = 5 * 60;
       if (
         TogglButton.$lastWork.id === TogglButton.$curEntry.id &&
-        inactiveSeconds >= 5 * 60
+        inactiveSeconds % FIVE_MINUTES === 0 ||
```

Comment:
This would only trigger exactly at 5 minutes or multiples of it, I don't that's the intended behaviour

Reply:
what would be the expected behaviour? @shantanuraj

Reply:
Checking for modulo assumes the callback will be executed precisely at the 5:00 minute mark
The callback is scheduled to exectue every 2 seconds, I think you should accord for some scheduling delays and accept a delta of at least +/- 2 seconds